src/engine/enginebuffer.cpp

```diff
@@ -595,29 +593,31 @@ void EngineBuffer::slotPassthroughChanged(double enabled) {
 }
 
 // WARNING: This method runs in both the GUI thread and the Engine Thread
-void EngineBuffer::slotControlSeek(double change) {
-    doSeek(change, SEEK_STANDARD);
+void EngineBuffer::slotControlSeek(double fractionalPos) {
+    doSeekFractional(fractionalPos, SEEK_STANDARD);
 }
 
 // WARNING: This method runs from SyncWorker and Engine Worker
-void EngineBuffer::slotControlSeekAbs(double abs) {
-    doSeek(abs / m_file_length_old, SEEK_STANDARD);
+void EngineBuffer::slotControlSeekAbs(double playPosition) {
+    doSeekPlayPos(playPosition, SEEK_STANDARD);
 }
 
 // WARNING: This method runs from SyncWorker and Engine Worker
-void EngineBuffer::slotControlSeekExact(double abs) {
-    doSeek(abs / m_file_length_old, SEEK_EXACT);
+void EngineBuffer::slotControlSeekExact(double playPosition) {
+    doSeekPlayPos(playPosition, SEEK_EXACT);
 }
 
-void EngineBuffer::doSeek(double change, enum SeekRequest seekType) {
+void EngineBuffer::doSeekFractional(double fractionalPos, enum SeekRequest seekType) {
     // Prevent NaN's from sneaking into the engine.
-    if (isnan(change)) {
+    if (isnan(fractionalPos)) {
         return;
     }
-
     // Find new playpos, restrict to valid ranges.
-    double new_playpos = round(change * m_file_length_old);
+    double new_playpos = round(fractionalPos * m_file_length_old);
```

Comment:
Forcing to even is already done at to places later in the code.

In a separate PR we should replace the double playpos by integer frames to avoid all these checks. 
In places where sample fractions make scene we should switch to double frames. This way we can express the position of a beat and we can deal with with scaling. 
But currently we cannot do this, since a position of 3 will be ironed to an even sample value. Once we have a position on stereo frames it will work.